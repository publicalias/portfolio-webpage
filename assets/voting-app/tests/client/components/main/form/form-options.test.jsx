"use strict";

//local imports

const FormOptions = require("../../../../../scripts/client/components/main/form/form-options");

const { testWrapper } = require("../../../test-helpers");

//global imports

const { initTestSnapshot } = require("redux/tests/client-tests");
const { reactTests } = require("redux/tests/react-tests");

//utilities

const { testShallow } = testWrapper(FormOptions);

//setup

beforeAll(reactTests.setup);
beforeEach(reactTests.inject(FormOptions, { lib: { chartColor: jest.fn(() => "black") } }));

//form options

describe("FormOptions", () => {

  const testSnapshot = initTestSnapshot(testShallow);

  it("should match snapshot (default)", () => testSnapshot());

  it("should match snapshot (options)", () => testSnapshot({ form: { body: { options: ["Option A"] } } }));

  it("should match snapshot (add)", () => testSnapshot({ form: { body: { add: "Option A" } } }));

});
