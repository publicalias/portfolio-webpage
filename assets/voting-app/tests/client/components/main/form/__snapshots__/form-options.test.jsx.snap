// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormOptions should match snapshot (add) 1`] = `
<div
  className="c-meta-poll-options"
>
  <MetaPollInput
    local={
      Object {
        "handleChange": [Function],
        "handleSubmit": [Function],
        "value": "Option A",
      }
    }
  />
</div>
`;

exports[`FormOptions should match snapshot (default) 1`] = `
<div
  className="c-meta-poll-options"
>
  <MetaPollInput
    local={
      Object {
        "handleChange": [Function],
        "handleSubmit": [Function],
        "value": "",
      }
    }
  />
</div>
`;

exports[`FormOptions should match snapshot (options) 1`] = `
<div
  className="c-meta-poll-options"
>
  <div
    className="u-margin-half"
  >
    <MetaPollOption
      key="0-730"
      local={
        Object {
          "created": true,
          "fill": "black",
          "handleRemove": [Function],
          "handleVote": [Function],
          "text": "Option A",
          "voted": false,
        }
      }
    />
  </div>
  <MetaPollInput
    local={
      Object {
        "handleChange": [Function],
        "handleSubmit": [Function],
        "value": "",
      }
    }
  />
</div>
`;
